api/filesys/fsondisk.go

```diff
@@ -100,6 +100,19 @@ func (fsOnDisk) IsDir(name string) bool {
 	return info.IsDir()
 }
 
+// ReadDir delegates to os.ReadDir
+func (fsOnDisk) ReadDir(name string) ([]string, error) {
+	dirEntries, err := os.ReadDir(name)
+	if err != nil {
+		return nil, err
+	}
+	var result []string
+	for _, d := range dirEntries {
+		result = append(result, d.Name())
+	}
```

Comment:
size of the result is known, so no need for the append
```
result := make(xx, len(dirEntries))
for i := range dirEntries {
  result[i] = dirEntries[i].Name
}
```

```

Reply:
done, thank you